src/main/java/com/openshift/jenkins/plugins/pipeline/dsl/OpenShiftBuilder.java

```diff
@@ -78,85 +81,127 @@ public String getCheckForTriggeredDeployments() {
-                Object commitID = arguments.get("commitID");
-                if (commitID != null) {
-                    step.setCommitID(commitID.toString());
-                }
-            }
-            if (arguments.containsKey("showBuildLogs")) {
-                Object showBuildLogs = arguments.get("showBuildLogs");
-                if (showBuildLogs != null) {
-                    step.setShowBuildLogs(showBuildLogs.toString());
-                }
-            }
-            if (arguments.containsKey("waitTime")) {
-                Object waitTime = arguments.get("waitTime");
-                if (waitTime != null) {
-                    step.setWaitTime(waitTime.toString());
-                }
-            }
-            ParamVerify.updateDSLBaseStep(arguments, step);
-            return step;
-        }
-    }
+	public static class DescriptorImpl extends AbstractStepDescriptorImpl implements IOpenShiftTimedStepDescriptor {
+		private String waitUnit = "sec";
+		private String wait;
+
+		public DescriptorImpl() {
+			super(OpenShiftBuilderExecution.class);
+		}
+
+		@Override
+		public String getFunctionName() {
+			return "openshiftBuild";
+		}
+
+		@Override
+		public String getDisplayName() {
+			return DISPLAY_NAME;
+		}
+
+		@Override
+		public Step newInstance(Map<String, Object> arguments) throws Exception {
+			if (!arguments.containsKey("buildConfig") && !arguments.containsKey("bldCfg"))
+				throw new IllegalArgumentException("need to specify buildConfig");
+			Object bldCfg = arguments.get("buildConfig");
+			if (bldCfg == null || bldCfg.toString().length() == 0)
+				bldCfg = arguments.get("bldCfg");
+			if (bldCfg == null || bldCfg.toString().length() == 0)
+				throw new IllegalArgumentException("need to specify buildConfig");
+			OpenShiftBuilder step = new OpenShiftBuilder(bldCfg.toString());
+			if(arguments.containsKey("buildName")) {
+				Object buildName = arguments.get("buildName");
+				if (buildName != null) {
+					step.setBuildName(buildName.toString());
+				}
+			}
+			if (arguments.containsKey("checkForTriggeredDeployments")) {
+				Object checkForTriggeredDeployments = arguments.get("checkForTriggeredDeployments");
+				if (checkForTriggeredDeployments != null) {
+					step.setCheckForTriggeredDeployments(checkForTriggeredDeployments.toString());
+				}
+			}
+			if (arguments.containsKey("commitID")) {
+				Object commitID = arguments.get("commitID");
+				if (commitID != null) {
+					step.setCommitID(commitID.toString());
+				}
+			}
+			if (arguments.containsKey("showBuildLogs")) {
+				Object showBuildLogs = arguments.get("showBuildLogs");
+				if (showBuildLogs != null) {
+					step.setShowBuildLogs(showBuildLogs.toString());
+				}
+			}
+
+			if (arguments.containsKey("waitTime")) {
+				Object waitTime = arguments.get("waitTime");
+				if (waitTime != null) {
+					wait = waitTime.toString();
+					step.setWaitTime(wait);
+				}
+			}
+
+			if (arguments.containsKey("waitUnit")) {
```

Comment:
leverage the new default method on IOpenShiftTimedStepDescriptor called for earlier